test/utils/RetryHelperSpec.scala

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import java.time.LocalDateTime
+import java.time.temporal.ChronoUnit
+
+import base.SpecBase
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.mockito.MockitoSugar
+import org.scalatest.time.{Seconds, Span}
+import org.scalatestplus.play.OneAppPerSuite
+import play.api.inject.bind
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.test.Helpers._
+import uk.gov.hmrc.http.{HttpException, Upstream5xxResponse}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class RetryHelperSpec extends SpecBase with MockitoSugar with ScalaFutures with OneAppPerSuite {
+
+  private val MAX_ATTEMPTS:Int = 10
+  private val INITIAL_WAIT:Int = 10
+  private val WAIT_FACTOR:Float = 1.5F
+
+  override lazy val app = new GuiceApplicationBuilder()
+    .configure(Map(
```

Comment:
is this really required for test? can we pass fakeConfig and remove the dependency of app and OneAppPerSuite

Reply:
Updated